parsing/cLexer.ml

```diff
@@ -548,42 +548,50 @@ let peek_marker s =
     | ('a'..'z' | 'A'..'Z' | '_') -> ImmediateAsciiIdent
     | _ -> raise Stream.Failure
 
-let parse_quotation loc s =
+let parse_quotation loc bp s =
   match peek_marker s with
   | ImmediateAsciiIdent ->
       let c = Stream.next s in
       let len =
         try ident_tail loc (store 0 c) s with
           Stream.Failure -> raise (Stream.Error "")
       in
-      get_buff len
+      get_buff len, set_loc_pos loc bp (Stream.count s)
   | Delimited (lenmarker, bmarker, emarker) ->
       let b = Buffer.create 80 in
       let commit1 c = Buffer.add_char b c; Stream.junk s in
       let commit l = List.iter commit1 l in
-      let rec quotation depth =
+      let rec quotation loc depth =
         match Stream.npeek lenmarker s with
         | l when l = bmarker ->
               commit l;
-              quotation (depth + 1)
+              quotation loc (depth + 1)
         | l when l = emarker ->
               commit l;
-              if depth > 1 then quotation (depth - 1)
+              if depth > 1 then quotation loc (depth - 1) else loc
+        | '\n' :: cs ->
+              commit1 '\n';
+              let loc = bump_loc_line_last loc (Stream.count s) in
+              quotation loc depth
         | c :: cs ->
               commit1 c;
-              quotation depth
+              quotation loc depth
         | [] -> raise Stream.Failure
       in
-      quotation 0;
-      Buffer.contents b
+      let loc = quotation loc 0 in
+      Buffer.contents b, set_loc_pos loc bp (Stream.count s)
 
 
-let find_keyword loc id s =
+let find_keyword loc id bp s =
```

Comment:
The main change is in the return type of this function that is now in charge of returning an updated loc, all the rest is propagation of this change.